RecoPixelVertexing/PixelVertexFinding/plugins/PixelVertexProducerFromSoA.cc

```diff
@@ -89,9 +87,12 @@ void PixelVertexProducerFromSoA::produce(edm::StreamID streamID, edm::Event &iEv
 
   int nv = soa.nvFinal;
 
-  // std::cout << "converting " << nv << " vertices " << " from " << indToEdm.size() << " tracks" << std::endl;
+#ifdef PIXVERTEX_DEBUG_PRODUCE
+  std::cout << "converting " << nv << " vertices " << " from " << indToEdm.size() << " tracks" << std::endl;
+#endif // PIXVERTEX_DEBUG_PRODUCE 
 
   std::set<uint16_t> uind;  // fort verifing index consistency
+  auto trackCollection = iEvent.getHandle(tokenTracks_);
```

Comment:
sorry, rereading the changes, isn't this the same as `tracks` at line 67 ?

Reply:
Indeed. 

We absolutely need a handle for making a TrackRef, as edm::Ref does not take references to product as a constructor parameter. It does does take pointers to product (and this compiles):
```c++
      //(old) auto tk = reco::TrackRef(trackCollectionHandle, k);
      auto tk = reco::TrackRef(&tracks, k);
```

but the constructor comments advise against using this to persist products which are in events (`DataFormats/Common/interface/Ref.h`):
```c++
    /// Constructors for ref to object that is not in an event.
    //  An exception will be thrown if an attempt is made to persistify
    //  any object containing this Ref.  Also, in the future work will
    //  be done to throw an exception if an attempt is made to put any object
    //  containing this Ref into an event(or run or lumi).
    Ref(product_type const* product, key_type itemKey, bool setNow = true);
```

So we can replace `track` with a handle and dereference it where needed (only one place, actually).

Fixed in latest commit, with an extra (unrelated) handle creation simplification.

Reply:
> It does take pointers to product (and this compiles)

Sure, but as you have seen in the comment that will give a transient-only `edm::Ref`, which does not know the original collection - so it's not what we want.